pkg/matcher/cel.go

```diff
@@ -24,28 +24,93 @@ const (
 	changedFilesTags = "files."
 )
 
+// evaluateBranchExpression function evaluates the branch expression for the given branch key from the provided expression.
+// Examples of expressions:
+//   - target_branch == "refs/heads*"
+//   - source_branch == "refs/heads*"
+//
+// If source_branch and target_branch contain anything other than "refs/heads*", the function won't perform any action.
+func evaluateBranchExpression(expression, branchKey string) bool {
```

Comment:
i am not understanding about the use of this func

Reply:
Today if cel expression has  `traget_branch` or `source_branch` with value `"refs/heads*"` it won't work

So basically this function is to evaluate the values of source_branch and target_branch if at all those contains  `"refs/heads*"`  then replace data map with `"refs/heads*"`

Let me know if this is still not clear